tools/toggle-fault-leds.hpp

```diff
@@ -0,0 +1,102 @@
+#pragma once
+#include "utility.hpp"
+
+/**
+ * @brief API to toggle fault LEDs according to the given functional status.
+ *
+ * The API sets "Functional" property hosted by operational status interface for
+ * all the FRUs under root path "/xyz/openbmc_project/inventory". It also
+ * fetches associated LEDs object path for FRUs, hosted under LED manager and
+ * sets asserted value for accordingly.
+ *
+ * The API skips some specific FRUs, although they host operational status based
+ * on following criteria.
+ * - Can be a guarded FRU.
+ * - Operational status of the FRU is handled by some other module.
+ *
+ * @param[in] isFunctional - States if the FRUs are functional or not.
+ */
+void toggleFaultLeds(const bool isFunctional)
+{
+    std::cout << "Trigger toggling of fault LEDs with value: " << isFunctional
+              << std::endl;
+
+    std::vector<std::string> skipOperationalStatusForFRUs{
+        "core", "powersupply", "unit", "connector", "chassis1"};
+
+    std::vector<std::string> frusWithoutLED{
+        "pcie_card", "usb",    "drive",       "ethernet",  "fan0_",
+        "fan1_",     "fan2_",  "fan3_",       "fan4_",     "fan5_",
+        "rdx",       "cables", "displayport", "pcieslot12"};
+
+    if (utility::isChassisOn())
+    {
+        std::cout << "Abort toggling fault LED. Chassis is in power on state."
+                  << std::endl;
+
```

Comment:
is empty line required ?

Reply:
Requirement can't be defined here. It makes the code looks clean and readable.